redshift/resource_redshift_grant.go

```diff
@@ -147,17 +154,22 @@ func resourceRedshiftGrantCreate(db *DBConnection, d *schema.ResourceData) error
 		return fmt.Errorf("Invalid privileges list %v for object of type %s", privileges, objectType)
 	}
 
+	databaseName := db.client.databaseName
```

Comment:
It seems to me like this logic is used multiple times. Would you be so kind to refactor the code, add a function and reduce duplicate code?  Thanks!